### PR TITLE
elk-#8 cirrus ci setup with custom docker image and aarch64 builds

### PR DESCRIPTION
**Pull request by elBoberido**
This PR adds a custom docker image for the CI and enables aarch64 builds

### CODE REVIEW

File: internal/docker/ubuntu-22.04 [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# Select base image
+FROM ubuntu:22.04
+
+# Disable prompt during packages installation
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Minimize the number of layers (RUN, COPY and ADD create layers)
+# https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers
+RUN echo "#### Installing dependencies" \
+  && apt update && apt install -y \
```
Comment (elfenpiff) [inline internal/docker/ubuntu-22.04:12]: For the `elkodon_env` I had to install `gcc` as well for bindgen - but when it works then it works :/

**Review by elfenpiff — COMMENTED** [review]
Is there a reason why the cirrus CI checks have vanished in this PR? They already saved my neck multiple times today.

File: internal/docker/ubuntu-22.04 (hunk 1 above)
Comment (elBoberido) [reply internal/docker/ubuntu-22.04:12]: The build error message requested clang and after adding it to the image the build was successful

File: .cirrus.yml [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+elk_common_cpu_and_memory_config_for_build_and_test_template: &ELK_COMMON_CPU_AND_MEMORY_CONFIG_FOR_BUILD_AND_TEST
+  cpu: 2
+  memory: 4GB
+
+elk_common_build_and_test_debug_template: &ELK_COMMON_BUILD_AND_TEST_DEBUG
+  env:
+    RUSTDOCFLAGS: -D warnings
+  timeout_in: 10m
+  build_script: cargo build --workspace --all --examples
```
Comment (elfenpiff) [inline .cirrus.yml:27]: I found out that benchmarks are also not included in `--all` could you add also `--benches` even when we do not yet have them. Then we cannot forget this in the future.

File: .cirrus.yml [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  doc_script: cargo doc
+  test_script: cargo test --workspace --all
+
+elk_common_build_and_test_release_template: &ELK_COMMON_BUILD_AND_TEST_RELEASE
+  env:
+    RUSTDOCFLAGS: -D warnings
+  timeout_in: 10m
+  build_script: cargo build --release --workspace --all --examples
```
Comment (elfenpiff) [inline .cirrus.yml:35]: Please add here `--benches` as well

File: .cirrus.yml [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  depends_on: ubuntu_22_04_x64_beta_release
+  allow_failures: true
+  container:
+    dockerfile: internal/docker/ubuntu-22.04
+    <<: *ELK_COMMON_CPU_AND_MEMORY_CONFIG_FOR_BUILD_AND_TEST
+  set_toolchain_script: rustup default nightly
+  <<: *ELK_COMMON_BUILD_AND_TEST_RELEASE
+
+#
+# Ubuntu aarch64
```
Comment (elfenpiff) [inline .cirrus.yml:102]: Woohoo :smile:

File: .cirrus.yml [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+  set_toolchain_script: rustup default beta
+  <<: *ELK_COMMON_BUILD_AND_TEST_RELEASE
+
+#
+# Windows Server 10 x86
+#
+
+# Pipeline 7
+
+windows_server_10_x64_stable_debug_task:
```
Comment (elfenpiff) [inline .cirrus.yml:189]: Awesome!

**Review by elfenpiff — CHANGES_REQUESTED** [review]
The commit message does not follow the elkodon guideline ... tsss tsss :grinning:

**Pull request merged**